### PR TITLE
Task#11009 Create Tabbar

### PR DESCRIPTION
**Pull request by huybnd-1817**
### Issue Link :link:
https://edu-redmine.sun-asterisk.vn/issues/11009

### Goals :soccer:

- Create TabBar with 2 items (Home, Shopping List)
- Go to the destination view
- Selected Tab and Unselected Tab

### Implementation Details :construction:

Tabbar2

### CODE REVIEW

File: CuisineRecipe/CuisineRecipe/App/AppDelegate.swift [hunk 1]
```diff
@@ -15,6 +15,9 @@ final class AppDelegate: UIResponder, UIApplicationDelegate {
     var window: UIWindow?
 
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+        //Setup Tabbar's properties
+        UITabBar.appearance().tintColor = UIColor(hexString: "#666666FF")
```
Comment (tienpx3010) [inline CuisineRecipe/CuisineRecipe/App/AppDelegate.swift]: Đưa vào func configTabbar

File: CuisineRecipe/CuisineRecipe/Extensions/UIColor+.swift [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+//  CuisineRecipe
+//
+//  Created by mac on 4/23/19.
+//  Copyright © 2019 sunasterisk. All rights reserved.
+//
+
+import UIKit
+
+extension UIColor {
+    public convenience init?(hexString: String) {
```
Comment (tienpx3010) [inline CuisineRecipe/CuisineRecipe/Extensions/UIColor+.swift]:
```suggestion
   convenience init?(hexString: String) {
```

File: CuisineRecipe/CuisineRecipe/App/AppDelegate.swift (hunk 1 above)
Comment (tanbtd-1747) [inline CuisineRecipe/CuisineRecipe/App/AppDelegate.swift]: đưa vào function configureTabBar()

File: CuisineRecipe/CuisineRecipe/Scenes/Storyboards/Base.lproj/Main.storyboard [hunk 3]
```diff
@@ -1,24 +1,106 @@
...
+                        <segue destination="xps-1N-EYN" kind="relationship" relationship="rootViewController" id="oGT-uT-RAR"/>
+                    </connections>
+                </navigationController>
+                <placeholder placeholderIdentifier="IBFirstResponder" id="xSH-X0-MZS" userLabel="First Responder" sceneMemberID="firstResponder"/>
+            </objects>
+            <point key="canvasLocation" x="1061.5999999999999" y="876.61169415292363"/>
         </scene>
     </scenes>
+    <resources>
+        <image name="home_icon" width="30" height="30"/>
```
Comment (tienpx3010) [inline CuisineRecipe/CuisineRecipe/Scenes/Storyboards/Base.lproj/Main.storyboard:103]: Ảnh đâu em nhỉ?

File: CuisineRecipe/CuisineRecipe/Extensions/UIColor+.swift [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        let r, g, b, a: CGFloat
+        
+        if hexString.hasPrefix("#") {
```
Comment (tanbtd-1747) [inline CuisineRecipe/CuisineRecipe/Extensions/UIColor+.swift]: dùng guard để giảm nesting level

File: CuisineRecipe/CuisineRecipe/Extensions/UIColor+.swift [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+            let start = hexString.index(hexString.startIndex, offsetBy: 1)
+            let hexColor = String(hexString[start...])
+            
+            if hexColor.count == 8 {
```
Comment (tienpx3010) [inline CuisineRecipe/CuisineRecipe/Extensions/UIColor+.swift]: Sai logic >= 6 < 6

File: CuisineRecipe/CuisineRecipe/Extensions/UIColor+.swift (hunk 4 above)
Comment (tienpx3010) [inline CuisineRecipe/CuisineRecipe/Extensions/UIColor+.swift]: Không cần # vẫn chạy nhé

File: CuisineRecipe/CuisineRecipe/App/AppDelegate.swift [hunk 6]
```diff
@@ -15,9 +15,16 @@ final class AppDelegate: UIResponder, UIApplicationDelegate {
     var window: UIWindow?
 
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+        configureTabBar()
         return true
     }
     
+    //Setup Tabbar's properties
```
Comment (tienpx3010) [inline CuisineRecipe/CuisineRecipe/App/AppDelegate.swift]: Bỏ

File: CuisineRecipe/CuisineRecipe/App/AppDelegate.swift [hunk 7]
```diff
@@ -15,9 +15,16 @@ final class AppDelegate: UIResponder, UIApplicationDelegate {
...
+    func configureTabBar() {
```
Comment (tienpx3010) [inline CuisineRecipe/CuisineRecipe/App/AppDelegate.swift]:
```suggestion
    func configTabBar() {
```

**Review by tienpx3010 — APPROVED** [review]

**Review by tanbtd-1747 — APPROVED** [review]

**Pull request merged**